code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -69,6 +73,43 @@
 		M.create_attack_log("<font color='orange'>Has had their soul captured with [src.name] by [key_name(user)]</font>")
 		user.create_attack_log("<font color='red'>Used the [src.name] to capture the soul of [key_name(M)]</font>")
 
+	if(optional)
+		if(!M.ckey)
+			to_chat(user, "<span class='warning'>They have no soul!</span>")
+			return
+		var/mob/player_mob = M
+		if(M.get_ghost())//in case our player ghosted and we need to throw the alert at their ghost instead
+			player_mob = M.get_ghost()
+		var/client/player_client = player_mob.client
+		to_chat(user, "<span class='warning'>You attempt to channel [M]'s soul into [src]. You must give the soul some time to react and stand still...</span>")
+		to_chat(player_mob, "<span class='warning'>[user] is trying to capture your soul into [src]! Click the button in the top right of the game window to respond.</span>")
+		var/obj/screen/alert/notify_soulstone/A = player_mob.throw_alert("\ref[src]_soulstone_thingy", /obj/screen/alert/notify_soulstone)
+		player_client << 'sound/misc/notice2.ogg'
+		window_flash(player_client)
+		if(player_client.prefs && player_client.prefs.UI_style)
+			A.icon = ui_style2icon(player_client.prefs.UI_style)
+		A.stone = src
+		A.stoner = user.real_name
+
+		var/old_layer = layer
+		var/old_plane = plane
+		layer = FLOAT_LAYER
+		plane = FLOAT_PLANE
+		A.overlays += src
+		layer = old_layer
+		plane = old_plane

```

Comment:
I don't think there's any reason to bother checking whether the player is in the body or not. Regular soulstones can still work on people outside of their body.

Also, the layer bit should probably be some sort of a sensible proc for assigning an atom as an overlay for the alert, but I'm not sure how implement that in a clean manner.